container-engine-lib/lib/backend_impls/docker/object_attributes_provider/object_attributes_provider.go

```diff
@@ -173,3 +188,67 @@ func (provider *dockerObjectAttributesProviderImpl) ForReverseProxy(engineGuid e
 	}
 	return objectAttributes, nil
 }
+
+// Return Traefik labels
+// Including the labels required to route traffic to the engine rest api port if the Host header is set to "engine".
+//
+//	"traefik.enable": "true",
+//	"traefik.http.routers.engine-rest-api.rule": "Host(`engine`)",
+//	"traefik.http.routers.engine-rest-api.service": "engine-rest-api",
+//	"traefik.http.services.engine-rest-api.loadbalancer.server.port": "<engine rest api port number>"
+func (provider *dockerObjectAttributesProviderImpl) getTraefikLabelsForEngine(restAPIPortSpec *port_spec.PortSpec) (map[*docker_label_key.DockerLabelKey]*docker_label_value.DockerLabelValue, error) {
```

Comment:
just a small improvement idea (probably coming from my functional oriented side :p) to deduce the code a bit. Basically create a map with the raw string pairs and then mapping over to the docker lables:
```
func (provider *dockerObjectAttributesProviderImpl) getTraefikLabelsForEngine(restAPIPortSpec *port_spec.PortSpec) (map[*docker_label_key.DockerLabelKey]*docker_label_value.DockerLabelValue, error) {
	var labelStringPairs map[string]string
	labelStringPairs = map[string]string{
		fmt.Sprintf("some %s", "key"): fmt.Sprintf("some %s", "value"),
		fmt.Sprintf("some %s", "key"): fmt.Sprintf("some %s", "value"),
		fmt.Sprintf("some %s", "key"): fmt.Sprintf("some %s", "value"),
	}

	labels := map[*docker_label_key.DockerLabelKey]*docker_label_value.DockerLabelValue{}
	for key, value := range labelStringPairs {
		labelKey, err := docker_label_key.CreateNewDockerTraefikLabelKey(key)
		if err != nil {
			return nil, stacktrace.Propagate(err, "An error occurred getting the traefik  label key with suffix '%v'", key)
		}
		labelValue, err := docker_label_value.CreateNewDockerLabelValue(value)
		if err != nil {
			return nil, stacktrace.Propagate(err, "An error occurred creating the traefik  label value with value '%v'", value)
		}
		labels[labelKey] = labelValue
	}

	return labels, nil
}
```

Reply:
I changed that and also for the user service labels.